src/CDF/CDFObject.php

```diff
@@ -52,13 +73,26 @@ class CDFObject implements CDFObjectInterface {
    * CDFObject constructor.
    *
    * @param string $type
+   *   Object type.
    * @param string $uuid
+   *   Object UUID.
    * @param string $created
+   *   Created date.
    * @param string $modified
+   *   Modified date.
    * @param string $origin
-   * @param array $metadata
-   */
-  public function __construct($type, $uuid, $created, $modified, $origin, $metadata = []) {
+   *   Origin UUID.
+   * @param array|mixed $metadata
+   *   Metadata.
+   */
+  public function __construct(
```

Comment:
Let's not do constructors this way. I think it's much more difficult to read and it doesn't match any standard we follow.